modules/failure-handler/src/main/java/org/apache/ignite/internal/failure/FailureProcessor.java

```diff
@@ -146,19 +161,58 @@ private synchronized boolean process(FailureContext failureCtx, FailureHandler h
         return invalidated;
     }
 
-    /**
-     * Returns {@code true} if the given failure type is ignored by the given handler.
-     *
-     * @param failureCtx Failure context.
-     * @param handler Handler.
-     */
-    private static boolean failureTypeIgnored(FailureContext failureCtx, FailureHandler handler) {
-        return handler instanceof AbstractFailureHandler
-                && ((AbstractFailureHandler) handler).ignoredFailureTypes().contains(failureCtx.type());
+    private void initFailureHandler() {
+        if (configuration == null) {
+            assert this.handler != null : "Failure handler is not initialized.";
+
+            return;
+        }
+
+        AbstractFailureHandler hnd;
+
+        FailureHandlerConfiguration handlerConfiguration = configuration.failureHandler();
+
+        switch (handlerConfiguration.failureHandlerType().value()) {
+            case NoOpFailureHandlerConfigurationSchema.TYPE:
+                hnd = new NoOpFailureHandler();
+                break;
+            case StopNodeFailureHandlerConfigurationSchema.TYPE:
+                hnd = new StopNodeFailureHandler();
+                break;
+            case StopNodeOrHaltFailureHandlerConfigurationSchema.TYPE:
+                StopNodeOrHaltFailureHandlerConfiguration stopNodeOrHaltConfiguration =
+                        (StopNodeOrHaltFailureHandlerConfiguration) handlerConfiguration;
+
+                boolean tryToStop = stopNodeOrHaltConfiguration.tryStop().value();
```

Comment:
Calling `stopNodeOrHaltConfiguration.value()` once and then accessing individual properties is a preferable way of doing it. Or even better - you could call `configuration.value()` in the beginning.